src/main/kotlin/it/pagopa/wallet/scheduler/repositories/WalletRepository.kt

```diff
@@ -0,0 +1,23 @@
+package it.pagopa.wallet.scheduler.repositories
+
+import it.pagopa.wallet.scheduler.documents.Wallet
+import org.springframework.data.mongodb.repository.Aggregation
+import org.springframework.data.repository.reactive.ReactiveCrudRepository
+import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+
+@Repository
+interface WalletRepository : ReactiveCrudRepository<Wallet, String> {
+
+    @Aggregation(
+        "{\$match: {'creationDate': {'\$gte': '?0','\$lte': '?1'},'status': '?2'}}",
```

Comment:
Since the creationDate of the last document processed by an iteration of the job becomes the startDate of the next iteration, consider using 'gt' instead of 'gte' while defining the match clause on creationDate

Reply:
Done here 778591f0e7d70ab35fdbc8052648e8b8baddb18e